### ISSUE TITLE
Session data model

### ISSUE THREAD

**Issue by argyleink** [GoogleChrome/devsummit#212]
curious what data we get from our sign in service about the user, and if we can append onto that object? so far, it looks like we're hoping to track these things about a user to help the ux/ui be contextually relevant:
- has not registered, registered but hasnt heard back, registered denied, registered accepted
- not attending, attending
- time zone
- favorited sessions []
- subscription type: push notifications &&|| emails
- sign in / sign out
- more??

ultimately, i'm trying to gather if there's enough custom profile / session information to necessitate a page devoted to managing it all. i'm leaning towards yes, but i'd like to avoid it if i can. thoughts? additions?

**Comment by surma** [comment]
? - has not registered, registered but hasnt heard back, registered denied, registered accepted
> - not attending, attending

we only get if they plan to attend, which is the button we've been talking about it #209 

> - time zone

we get that from the browser

> - favorited sessions []

yes

> - subscription type: push notifications &&|| emails

I think for now we are doing push notifications only

> - sign in / sign out
> - more??

Avatar. That's it for now.

**Comment by argyleink** [comment]
i'm feeling conflicting ux desires:
- user should be able to freely toggle their attending state
- registration provides a boolean regarding attending

which one wins here? we dont need both, at least not in the UI. 

seems like we're not tracking enough to devote a page. i'll try and stick these personalizable elements in context.

**Comment by argyleink** [comment]
- user should be able to wipe all data we've stored on them

**Comment by surma** [comment]
> registration provides a boolean regarding attending

We don't get any data from defjam. So we only have the toggle. Nothing else.

**Comment by jakearchibald** [comment]
> i'm feeling conflicting ux desires:
> 
> * user should be able to freely toggle their attending state
> * registration provides a boolean regarding attending
> 
> which one wins here? we dont need both, at least not in the UI.

We're not connected up to defjam so we can't query the state of existence of the user's attendance ticket.

All we offer is a toggle for the user experience that's optimised for either a live streamer, or an attendee.

If we think this is going to be tough UX-wise, we could avoid having the switch until the event is happening. Then the switch can be "I'm attending!", which isn't as easily confused with registration.

**Comment by surma** [comment]
> we could avoid having the switch until the event is happening

I like this. At that point we could call the button "I am at the venue!" or something which removes all possible confusion that the toggle might mean giving up your ticket or something.

**Comment by jakearchibald** [comment]
> "I am at the venue!"

I thought the same, but I wondered about cases where they're attending, but not at the venue (eg, trying to find the venue). But maybe that information will be easy to find anyway.

**Comment by argyleink** [comment]
i like that idea as well, more contextually relevant and will prevent confusion. cool! i'll design the user profile page to have the switch always, but what do you think of a conditionally appearing module for "i am at the venue" notice and action?

you know, at that point, it's almost like you could suggest they're attending or just switch the state of the site, if their GPS is within the geo fence of the event. would make this a more "intelligent" feature. 

"the site changed to an attending state when i allowed geolocation and was at the conf!"

label it nice to have for sure, but would be neat. lol, insert jake "this is feature driven design" which i'd retort it was discovered by thinking about the user 😂 

**so to recap current state of threads on here:**
- we're not getting a callback or making any assumptions about defjam state, we'll manage our own attendance bool
- we're making a page #224 to hold the information and actions the user can take on it

surma or jake, since this ticket is about the data model we'll be holding for user settings, can y'all share with me what our current belief is around the shape of that model? 

**here's my stab:**
- not attending, attending
- time zone
- favorited sessions []
- push notification permission state?
- signed in or not

**actions for the user:**
- delete data
- change timezone
- switch attending/not
- sign out / sign in

**Comment by argyleink** [comment]
i'm also curious how much we want to gate behind auth and what we can stash in localstorage without auth? 

like, we could have all clicks of the avatar icon or profile avatar go to this settings page, aka not gated. let them login from that page, tell them what they get if/when they login, and yeah, simplify yet centralize.

**Comment by jakearchibald** [comment]
> but what do you think of a conditionally appearing module for "i am at the venue" notice and action?

Yeah, sounds good.

> you know, at that point, it's almost like you could suggest they're attending or just switch the state of the site, if their GPS is within the geo fence of the event.

I guess we could avoid showing "I am at the venue" if the IP is nowhere near. I worry a little about reliability there.

I don't think we should use accurate GPS. We're trying to answer the question, "Are you at the venue?", so asking the user "Can we have permission to read your location?" is a step removed. It's also privacy invasive, as we don't need to know their precise location if they're in another country. It's a cool tech demo, but I don't think it's the best UX.

GPS would be a good think to offer for the viewing parties page though. "Find my closest viewing party".

> insert jake "this is feature driven design"

Haha I wrote the above before seeing this. You got me.

> surma or jake, since this ticket is about the data model we'll be holding for user settings, can y'all share with me what our current belief is around the shape of that model?

* Attending?
* Is speaker? (maybe. It's something we captured in the doc anyway)
* Favorited sessions [].
* Push notification details.

I reckon.

> i'm also curious how much we want to gate behind auth and what we can stash in localstorage without auth?

I think we can use origin storage as a cache, but update from the server at regular intervals.
